facade/publicendpoint.go

```diff
@@ -232,3 +232,73 @@ func (f *Facade) EnablePublicEndpointPort(ctx datastore.Context, serviceid, endp
 	glog.V(2).Infof("Service (%s) updated", svc.Name)
 	return nil
 }
+
+// Adds a vhost public endpoint to a service
+func (f *Facade) AddPublicEndpointVHost(ctx datastore.Context, serviceid, endpointName, vhostName string, isEnabled, restart bool) (*servicedefinition.VHost, error) {
+	// Get the service for this service id.
+	svc, err := f.GetService(ctx, serviceid)
+	if err != nil {
+		err = fmt.Errorf("Could not find service %s: %s", serviceid, err)
+		glog.Error(err)
+		return nil, err
+	}
+
+	// check other virtual hosts for redundancy
+	vhostLowerName := strings.ToLower(vhostName)
+	if services, err := f.GetAllServices(ctx); err != nil {
+		err = fmt.Errorf("Could not get the list of services: %s", err)
+		glog.Error(err)
+		return nil, err
+	} else {
```

Comment:
Without the else, services has to be declared outside of the if() statement.  I can do that.